include/IECoreScene/private/PrimitiveAlgoUtils.h

```diff
@@ -184,15 +184,15 @@ class SplitTask : public tbb::task
 	private:
 		typedef typename P::Ptr Ptr;
 	public:
-		SplitTask(const std::vector<T> &segments, P *primitive, const S& splitter, const std::string &primvarName, std::vector<Ptr> &outputPrimitives, size_t offset, size_t depth = 0)
+		SplitTask(const std::vector<T> &segments, typename P::Ptr primitive, const S& splitter, const std::string &primvarName, std::vector<Ptr> &outputPrimitives, size_t offset, size_t depth = 0)
```

Comment:
I'd like to understand the need for this. In `execute()` below, I would assume that `Ptr a`, `Ptr b` and `wait_for_all()` guarantee that the primitives are alive for longer than the child tasks are. And I would guess that the client of TaskSegmenter is likewise required to keep the outer primitive alive for the duration. What am I missing?

Reply:
I write them into an array for use outside when they're not going to be split further:

https://github.com/ImageEngine/cortex/blob/ea22bd6aa61b19f6d692dba13b67af58f60540a9/include/IECoreScene/private/PrimitiveAlgoUtils.h#L197